layers/adblock.go

```diff
@@ -31,18 +31,18 @@ type AdblockLayer struct {
 	cond    *sync.Cond
 }
 
-func (a *AdblockLayer) OnRequest(state *httransform.LayerState) error {
-	host, _ := state.RequestHeaders.GetString("host")
-	contentType, _ := state.RequestHeaders.GetString("content-type")
-	referer, _ := state.RequestHeaders.GetString("referer")
+func (a *AdblockLayer) OnRequest(ctx *layers.Context) error {
+	host := ctx.RequestHeaders.GetFirst("host").Value()
```

Comment:
In general, I would recommend to take last header instead of the first one. The vast majority of clients append key/value pairs because of the agreement that 'last header wins'

Reply:
Thank you.